File: src/slotrenderer.cpp
#include "slotrenderer.h"

#include "glgraphics.h"

SlotRenderer::SlotRenderer()
{
}

void SlotRenderer::render(const Slot &slot) const
{
	Color slotColor = slot.getColor();
	vec2f slotCenter = slot.getCenter();

	GLGraphics *glGraphics = GLGraphics::getInstance();

	glGraphics->setColor(slotColor);
	glGraphics->drawCircle(slotCenter, 0.05, true);
}

File: src/js/slotrenderer.js
class SlotRenderer
{
	constructor(ctx)
	{
		this.graphics = graphics
	}

	function render(slot)
	{
		var slotColor = slot.getColor();
		var slotCenter = slot.getCenter();
		var radius = 20;

		this.graphics.setColor(slotColor);
		this.graphics.drawCircle(slotCenter, radius, true);
	}
}
